centrych_os_probe.patch
Description: Modified grub menu generation.  Looks for Centrych partitions and
 places drive in brackets.
Author:      Jack Radigan <jprad@centrych.org>
Last-Update: 2013-08-04
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -31,7 +31,8 @@
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
 else
-  if [ "${GRUB_DISTRIBUTOR}" = "Ubuntu" ] ; then
+  if [ "${GRUB_DISTRIBUTOR}" = "Ubuntu" ] || \
+    [ "${GRUB_DISTRIBUTOR}" = "Centrych" ] ; then
     OS="${GRUB_DISTRIBUTOR}"
   else
     OS="${GRUB_DISTRIBUTOR} GNU/Linux"
@@ -103,6 +104,11 @@
   if [ -z "$boot_device_id" ]; then
       boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
   fi
+  if [ -z "$submenu_indentation" ]; then
+      boot_device_label=" [$GRUB_DEVICE]"
+  else
+      boot_device_label=""
+  fi
   if [ x$type != xsimple ] ; then
       case $type in
 	  recovery)
@@ -116,9 +122,9 @@
 	  title_correction_code="${title_correction_code}if [ \"x\$default\" = '$quoted' ]; then default='$(echo "$replacement_title" | grub_quote)'; fi;"
 	  grub_warn "$(gettext_printf "Please don't use old title \`%s' for GRUB_DEFAULT, use \`%s' (for versions before 2.00) or \`%s' (for 2.00 or later)" "$GRUB_ACTUAL_DEFAULT" "$replacement_title" "gnulinux-advanced-$boot_device_id>gnulinux-$version-$type-$boot_device_id")"
       fi
-      echo "menuentry '$(echo "$title" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-$version-$type-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
+      echo "menuentry '$(echo "$title" | grub_quote)$boot_device_label' ${CLASS} \$menuentry_id_option 'gnulinux-$version-$type-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
   else
-      echo "menuentry '$(echo "$os" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-simple-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
+      echo "menuentry '$(echo "$os" | grub_quote)$boot_device_label' ${CLASS} \$menuentry_id_option 'gnulinux-simple-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
   fi      
   echo "recordfail" | sed "s/^/$submenu_indentation/"
   if [ x$type != xrecovery ] ; then
@@ -323,7 +329,7 @@
 	boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
     fi
     # TRANSLATORS: %s is replaced with an OS name
-    echo "submenu '$(gettext_printf "Advanced options for %s" "${OS}" | grub_quote)' \$menuentry_id_option 'gnulinux-advanced-$boot_device_id' {"
+    echo "submenu '$(gettext_printf "Advanced options for %s" "${OS}" | grub_quote) [$GRUB_DEVICE]' \$menuentry_id_option 'gnulinux-advanced-$boot_device_id' {"
   fi
 
   linux_entry "${OS}" "${version}" advanced \
--- a/util/grub.d/30_os-prober.in
+++ b/util/grub.d/30_os-prober.in
@@ -108,7 +108,7 @@
 	bitstr="$(gettext "(64-bit)")"
     fi
     # TRANSLATORS: it refers on the OS residing on device %s
-    onstr="$(gettext_printf "(on %s)" "${DEVICE}")"
+    onstr="$(gettext_printf "[%s]" "${DEVICE}")"
         cat << EOF
 menuentry '$(echo "${LONGNAME} $bitstr $onstr" | grub_quote)' --class osx --class darwin --class os \$menuentry_id_option 'osprober-xnu-$2-$(grub_get_device_id "${DEVICE}")'  {
 EOF
@@ -200,7 +200,7 @@
       esac
 
       found_other_os=1
-	  onstr="$(gettext_printf "(on %s)" "${DEVICE}")"
+	  onstr="$(gettext_printf "[%s]" "${DEVICE}")"
       cat << EOF
 menuentry '$(echo "${LONGNAME} $onstr" | grub_quote)' --class windows --class os \$menuentry_id_option 'osprober-chain-$(grub_get_device_id "${DEVICE}")' {
 EOF
@@ -253,7 +253,7 @@
 	fi
 
 	found_other_os=1
-	onstr="$(gettext_printf "(on %s)" "${DEVICE}")"
+	onstr="$(gettext_printf "[%s]" "${DEVICE}")"
 	recovery_params="$(echo "${LPARAMS}" | grep 'single\|recovery')" || true
 	counter=1
 	while echo "$used_osprober_linux_ids" | grep 'osprober-gnulinux-$LKERNEL-${recovery_params}-$counter-$boot_device_id' > /dev/null; do
@@ -266,7 +266,7 @@
 
 	if [ "x$is_first_entry" = xtrue ]; then
             cat << EOF
-menuentry '$(echo "$OS" | grub_quote)' --class gnu-linux --class gnu --class os \$menuentry_id_option 'osprober-gnulinux-simple-$boot_device_id' {
+menuentry '$(echo "$OS" | grub_quote) $onstr' --class gnu-linux --class gnu --class os \$menuentry_id_option 'osprober-gnulinux-simple-$boot_device_id' {
 EOF
 	    save_default_entry | sed -e "s/^/\t/"
 	    printf '%s\n' "${prepare_boot_cache}"
@@ -281,10 +281,10 @@
         cat << EOF
 }
 EOF
-	    echo "submenu '$(gettext_printf "Advanced options for %s" "${OS}" | grub_quote)' \$menuentry_id_option 'osprober-gnulinux-advanced-$boot_device_id' {"
+	    echo "submenu '$(gettext_printf "Advanced options for %s" "${OS}" | grub_quote) $onstr' \$menuentry_id_option 'osprober-gnulinux-advanced-$boot_device_id' {"
 	    is_first_entry=false
 	fi
-	title="${LLABEL} $onstr"
+	title="${LLABEL}"
         cat << EOF
 	menuentry '$(echo "$title" | grub_quote)' --class gnu-linux --class gnu --class os \$menuentry_id_option 'osprober-gnulinux-$LKERNEL-${recovery_params}-$boot_device_id' {
 EOF
@@ -320,7 +320,7 @@
     ;;
     hurd)
       found_other_os=1
-      onstr="$(gettext_printf "(on %s)" "${DEVICE}")"
+      onstr="$(gettext_printf "[%s]" "${DEVICE}")"
       cat << EOF
 menuentry '$(echo "${LONGNAME} $onstr" | grub_quote)' --class hurd --class gnu --class os \$menuentry_id_option 'osprober-gnuhurd-/boot/gnumach.gz-false-$(grub_get_device_id "${DEVICE}")' {
 EOF
